src/compile-middleware.ts

```diff
@@ -42,66 +46,71 @@ function isSuccessful(response: Response) {
   return (statusCode >= 200 && statusCode < 300);
 }
 
-export const babelCompile: RequestHandler = transformResponse({
-  shouldTransform(request: Request, response: Response) {
-    return isSuccessful(response) &&
-        compileMimeTypes.indexOf(getContentType(response)) >= 0;
-  },
-
-  transform(request: Request, response: Response, body: string): string {
-    const contentType = getContentType(response);
-    const uaParser = new UAParser(request.headers['user-agent']);
-    const compile = needCompilation(uaParser);
-
-    if (compile) {
-      // TODO(justinfagnani): cache compilation results
-      if (contentType === htmlMimeType) {
-        const document = parse5.parse(body);
-        const scriptTags = dom5.queryAll(document, isInlineJavaScript);
-        for (const scriptTag of scriptTags) {
+export function babelCompile(forceCompile: boolean): RequestHandler {
+  if (forceCompile == null)
+    forceCompile = false;
+
+  return transformResponse({
+    shouldTransform(request: Request, response: Response) {
+      return isSuccessful(response) &&
+          compileMimeTypes.indexOf(getContentType(response)) >= 0;
+    },
+
+    transform(request: Request, response: Response, body: string): string{
+      const contentType = getContentType(response);
+      const uaParser = new UAParser(request.headers['user-agent']);
+      const compile = forceCompile || needCompilation(uaParser);
+
+      if (compile) {
+        // TODO(justinfagnani): cache compilation results
```

Comment:
For development it may be that the built-in HTTP caching that we're getting with send() should make this just fine. It won't help at all with multiple clients, but it'll work fine for a single client.